Port SecretCache to task sdk

prabhusneha opened the pull request:
Closes: https://github.com/apache/airflow/issues/48833

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

rawwar committed:
capitalize the term airflow (#49450)

Commit:
Port SecretCache to task sdk

Commit:
Port SecretCache to task sdk

kevinhongzl committed:
docs: fix broken link to testing doc (#49442)

eladkal committed:
Prepare docs for Apr ad hoc release of edge3 and git providers (#49465)

vincbeck committed:
Make AWS auth manager compatible with AF3 (#49419)

cmarteepants committed:
Add comprehensive release notes for Airflow 3.0.0 (#49468)

This PR adds the full release notes for Apache Airflow 3.0.0, covering:

- Major architectural changes (Task SDK, Task Execution API, Edge Executor)
- Native DAG versioning and asset-based scheduling
- UI overhaul with a new React-based frontend
- DAG authoring improvements, including support for ML and event-driven workflows
- Updated config defaults, removed features, and breaking changes
- Scheduler and executor behavior updates
- CLI and REST API changes
- Provider refactoring and standardization
- Migration tools and upgrade guidance

**Note:** Some changes span multiple categories. For example, catchup_by_default is a global configuration change, but it also impacts DAG authoring behavior. In such cases, the change is referenced in more than one section to provide appropriate context for different audiences.

This reflects months of community contributions and the culmination of a long-standing roadmap toward a more modular, scalable, and API-driven Airflow.

jscheffl committed:
Ensure fab provider is installed when running EdgeExecutor (#49473)

guan404ming committed:
AIP-38: set default sort by key (#49480)

guan404ming committed:
AIP-38: set default sort by key in `Connections` (#49481)

bugraoz93 committed:
Fix path checking for credentials with exact credential path (#49444)

dheerajturaga committed:
Bring back support for local dagbag based list/list-import-errors (#49380)

* Bring back support for localdagbag based list/list-imporr-errors

* fix typo

* Fix unit tests - restore db back to its orig state for downstream tests

* Absorb Ruff linting to fix CI static checks

* better help message

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

---------

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

uranusjr committed:
Mention key changes in event accessors (#49491)

amoghrajesh committed:
Remove flask application configuration from docs for AF3 (#49393)

uranusjr committed:
Better typing in asset event accessors (#49488)

uranusjr committed:
Change wording (#49493)

eladkal committed:
Update providers metadata 2025-04-19 (#49494)

dheerajturaga committed:
Add aarch64 -> arm64 platform normalization in breeze (#49484)

This is needed so that some breeze commands can be supported on aarch machines.

TJaniF committed:
Correct asset event listener typo (#49485)

bugraoz93 committed:
Breeze release_management_commands use already passed init_file_path (#49443)

guan404ming committed:
AIP-38: merge Backfill and Trigger Dag Run (#49490)

amoghrajesh committed:
Move DagIsPaused exception to standard provider (#49500)

* Move DagIsPaused exception to standard provider

* remove from core

dheerajturaga committed:
airflow-cli to depend on airflow common services (#49318)

Several airflow cli commands would fail if the airflow common dependencies are removed. Adding  airflow-common-depends-on

simonprydden committed:
fix: correct invalid arg timeout (#49426)

Co-authored-by: Simon Prydden <mea99sd@aol.com>

aritra24 committed:
Add airflowctl codegen (#48914)

* Add airflowctl codegen

* Added the generated file

* Adds license generated.py

Moved datamodel_code_formatter to devel-common
and updated pyproject.toml

* Create simple extensible formatter

Currently only takes mypy ignores to append
to the file. Unfortunately datamodel-codegen
only supports a single string so would have to
put in all the ignores into one string if there's
multiple.

* Undid the custom codegen changes

* Excluded the generated files from mypy

* Reverted py3.10 change and updated change monitor files

* Changed name of the yaml file

* Push new generated files

* Ignore generated files from add license

* regenerated generated.py

* Updated .rat-excludes for gen files

guan404ming committed:
feat: handel remove dags (#49504)

guan404ming committed:
chore: import paths use the stable functions (#49460)

* chore: update deprecated functions to use the stable functions

* env: bump common.sql to `1.21.0`

dstandish committed:
Remove dag_code records with no serdag (#49478)

It is assumed by the migration process that there will not be any dag_code records that don't have a serdag record.
It seems this only happens when there are two files with the same dag id.

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

jscheffl committed:
Update screenshots for Airflow 3 UI overview (#49518)

* Update screenshots for Airflow 3 UI overview

* Filename-typo

jscheffl committed:
Fix static checks on airflowctl generated client code (#49523)

simonprydden committed:
feat: add expire_time configuration to OracleHook connection (#49521)

Co-authored-by: Simon Prydden <mea99sd@aol.com>

guan404ming committed:
enhance: logs SQL before execution in `snowflake` and `databricks_sql` (#48942)

jscheffl committed:
Update screenshots for Airflow 3 Architecture (#49524)

jscheffl committed:
Update screenshots for Dynamic Task Mapping (#49529)

jscheffl committed:
Update screenshots for Asset Scheduling (#49530)

jscheffl committed:
Update screenshots for Dag-Run (#49528)

rawwar committed:
make `private_key_content` in snowflake connection to be a base64 encoded string (#49467)

* decode private key using base64

* update docs

* update tests

* update changelog

* update version in changelog

* update changelog to include how to migrate

* remove major version update from changelog

* update changelog

* Update providers/snowflake/docs/changelog.rst

---------

Co-authored-by: Elad Kalif <45845474+eladkal@users.noreply.github.com>

Commit:
Remove duplicate 'mock_supervisor_comms' pytest fixtures from core (#49457)

jscheffl committed:
Update screenshots for Dags (#49527)

* Update screenshots for Dags

* Convert mp4 to gif

---------

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

jedcunningham committed:
Remove `gitpython` as a core dependency (#49537)

This was added when `GitDagBundle` lived in core, but it now lives in
the git provider. We no longer need gitpython as a core dependency.

Commit:
Delete duplicate `mock_supervisor_comms` pytest fixtures from OL provider (#49520)

* Remove duplicate mock_supervisor_comms pytest fixture definitions

* Remove duplicate mock_supervisor_comms pytest fixture definitions

* Revert

* Remove mock_supervisor_comms definition from conftest

* Remove duplicate mock_supervisor_comms definition from provider

* Fix static check

---------

Co-authored-by: Sneha Prabhu <snehaprabhu@Snehas-MacBook-Pro.local>
Co-authored-by: Amogh Desai <amoghrajesh1999@gmail.com>

cmarteepants committed:
update ui guide for 3.0 (#49532)

* Update ui guide for 3.0

This PR expands the Airflow UI guide with the following new sections:
	•	DAG Run View: Includes task-level metadata, code snapshotting, and visual debugging via Graph View
	•	Task Instance Details: Covers log inspection, error tracing, rendered templates, and XComs
	•	Asset Views: Introduces the Assets tab, asset graphs, event tracking, and asset-triggered DAGs
	•	Admin Views: Documents key features in Variables, Pools, Connections, Plugins, Providers, and Config

Also includes:
	•	Screenshot callouts for both light and dark modes
	•	Minor formatting and nesting cleanup to improve readability

Related: #45649

* Rearrange dir for screenshots

---------

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

oboki committed:
Add `STRAIGHT_JOIN` prefix for MySQL query optimization in get_sorted_triggers (#46303)

This PR adds a STRAIGHT_JOIN hint to prevent an unintended full scan of TaskInstance table in get_sorted_triggers.

With MySQL backend, if there are many records in the trigger, it causes the task_instance to be scanned first as the driving table, leading to slow queries.

amoghrajesh committed:
Adding change in `xcom_pull` behaviour to release notes (#49455)

amoghrajesh committed:
Update the doc refs for 'asset-scheduling' (#49539)

kaxil committed:
Add note on XCom pull behaviour change (#49546)

related: https://github.com/apache/airflow/issues/49540

kaxil committed:
Fix failing doc tests on main (#49551)

Example failure: https://github.com/apache/airflow/actions/runs/14595033061/job/40939162704?pr=49549

```
 WARNING: unknown document: 'apache-airflow:authoring-and-scheduling/datasets'

File path: /opt/airflow/providers-summary-docs/core-extensions/asset-schemes.rst:27

  22 | exposed via Airflow core and community-managed providers.
  23 |
  24 | ..
  25 |   TODO: Change this from Dataset to Asset in Airflow 3.0
  26 |
> 27 | See :doc:`documentation on Assets <apache-airflow:authoring-and-scheduling/datasets>` for details on how asset URIs work.
  28 |
  29 | .. airflow-dataset-schemes::
  30 |    :tags: None
  31 |    :header-separator: "

```

kaxil committed:
Fix broken spell check (#49552)

```
RELEASE_NOTES.rst:345: (im)  Airflow 3 resolves im inconsistency by requiring

File path: /opt/airflow/airflow-core/docs/RELEASE_NOTES.rst
Incorrect Spelling: 'im'
Line with Error: 'Airflow 3 resolves im inconsistency by requiring '
====================================================================================================

If there are spelling errors related to class or function name, make sure
those names are quoted with backticks '`' - this should exclude it from spellcheck process.
If there are spelling errors in the summary above, and the spelling is
correct, add the spelling to docs/spelling_wordlist.txt or use the
spelling directive.
Check https://sphinxcontrib-spelling.readthedocs.io/en/latest/customize.html#private-dictionaries
for more details.

If there are no spelling errors in the summary above, there might be an
issue unrelated to spelling. Please review the traceback.

```

potiuk committed:
Disable sdist verification for now (#49553)

Until we find the reason why setuptools-scm fails building wheels
from sdist, we disable it.

potiuk committed:
Fix no-provider constraint generation (#49549)

dependabot[bot] committed:
Bump @babel/runtime (#49479)

Bumps [@babel/runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime) from 7.26.0 to 7.27.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime)

---
updated-dependencies:
- dependency-name: "@babel/runtime"
  dependency-version: 7.27.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

potiuk committed:
Revert "Disable sdist verification for now (#49553)" (#49555)

This reverts commit e255782f020fa656075cc8b408d5641bb9d4c4d4.
ands removes --no-binary :all: that caused failure with setuptools-scm

potiuk committed:
Get rid of chicken-egg-providers AFTER we release Airflow 3.0.0 (#49501)

Since we are starting RC candidates from 1 - we should also produce
them as RC1 when chicken-egg providers are used to build
images from main - this will correctly skip them when they are
already released. We should also make sure to merge and remove all the
chicken-egg-providers now when all of them are already relased and
when Airflow 3.0.0 is released as well. This will bring "reason" back
to building the images.

guan404ming committed:
AIP-38: align delete model (#49514)

pierrejeambrun committed:
Remove sort indicators on XCom table (#49547)

rawwar committed:
use Boolean (#49556)

kaxil committed:
Bump Airflow to `3.1.0.dev0` (#49119)

Airflow main is now targeting 3.1 with the final rc

ellisms committed:
Amazon EMR Sensors/Operators inherit AWS Base Classes (#49486)

* EMR operators inherit AwsBaseOperator

* Update EMR sensors to inherit AwsBaseSensor

---------

Co-authored-by: mse139 <mse139@users.noereply.github.com>

amoghrajesh committed:
Improve docs-build selective checks to include RELEASE_NOTES.rst (#49558)

vincbeck committed:
Fix AWS auth manager system test (#49561)

dependabot[bot] committed:
Bump the fab-ui-package-updates group across 1 directory with 2 updates (#49511)

Bumps the fab-ui-package-updates group with 2 updates in the /providers/fab/src/airflow/providers/fab/www directory: [eslint](https://github.com/eslint/eslint) and [webpack](https://github.com/webpack/webpack).


Updates `eslint` from 9.24.0 to 9.25.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.24.0...v9.25.0)

Updates `webpack` from 5.99.5 to 5.99.6
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.99.5...v5.99.6)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.25.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: fab-ui-package-updates
- dependency-name: webpack
  dependency-version: 5.99.6
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: fab-ui-package-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

potiuk committed:
Add backwards compatibility provider tests for Airflow 3.0.0 (#49562)

shubhamraj-git committed:
add code owner (#49578)

tirkarthi committed:
Vertically align task log header components in fullscreen mode. (#49569)

Co-authored-by: Shubham Raj <48172486+shubhamraj-git@users.noreply.github.com>

dstandish committed:
Ensure sqlalchemy[asyncio] extra is in core deps (#49452)

This is required for async sqlalchemy operations.  This surfaces e.g. when we install with `--no-dev` and run the api-server, we'll get missing module `greenlet` error in api-server logs.

dstandish committed:
Null out the scheduler_interval field on downgrade (#49583)

In 2.x, it was expected that this field was valid json -- e.g. `"0 0 * * *"`.

In 3.x, the column is renamed and repurposed and not json.

So if we downgrade it, we need to null it out.  It will get updated when dag is reserialized.

If we *don't* do this, the scheduler crashes.

dstandish committed:
Remove unused constant HANDLER_SUPPORTS_TRIGGERER (#49370)

This was obsoleted by logging changes in task sdk, but its removal was missed, it seems.

kaxil committed:
Update Airflow version to 3.0.0 in README & other docs (#49585)

Cherry-picking this from v3-0 branches back to main.

Co-authored-by: Jed Cunningham <jedcunningham@apache.org>

jscheffl committed:
Update screenshots for Dags - Update GIFs (#49586)

guan404ming committed:
docs: add README.md in `airflow-ctl` (#49464)

vincbeck committed:
Fix AWS auth manager (#49588)

kaxil committed:
Reorder versions for Bug Reports (#49592)

Instead of 2.10.5 as first optoin, we default it to 3.0.0

guan404ming committed:
refactor: use common variable in ci (#49463)

o-nikolas committed:
Fix typo in emr sensor (#49597)

amoghrajesh committed:
Logout functionality should respect base_url in api server (#49545)

amoghrajesh committed:
Updating Airflow executor docs for AF3 (#49389)

kaxil committed:
Add Code Coverage for Airflow 3 branches (#49595)

kaxil committed:
Update Badge links for Task SDK Readme (#49589)

kaxil committed:
Make `v3-0-stable` a protected branch (#49594)

kacpermuda committed:
docs: Add provider versions to supported_classes OL doc (#49409)

amoghrajesh reviewed (commented):
Generally the direction for this is ok. 

1. I am not sure if moving it to a "cache" is needed however, we can just have it be in execution_time instead: `task-sdk/src/airflow/sdk/execution_time`. You can have a cache there, it wont really matter.
2. Can you please add a newsfragment for this (i mean its not a user facing one so it can be a significant one but need not have migration rules)
3. Add a lazy import like https://github.com/apache/airflow/pull/46176/files#diff-d452a01c463a734b311353185dc955ac7b80c4e2bf00533f430f094dc06a3402R51-R52 so we can use it like: `from airflow.sdk import SecretCache`

amoghrajesh committed:
Adding flask app configuration docs to FAB provider (#49492)

* Adding flask app configuration docs to FAB provider

* updating it

* Apply suggestions from code review

Co-authored-by: Vincent <97131062+vincbeck@users.noreply.github.com>

---------

Co-authored-by: Vincent <97131062+vincbeck@users.noreply.github.com>

amoghrajesh committed:
Introduce lower bind to lxml as 5.4.0 (#49612)

amoghrajesh committed:
Fixing upgrade checks on main (#49610)

* Fixing upgrade checks on main

* reuse variable

potiuk committed:
Actually fail when number of attempts failed for breeze command (#49615)

kacpermuda committed:
tests: Fix OpenLineage VariableTransport's initialization (#49550)

vatsrahul1001 committed:
Update docs to ensure Airflow 2 plugins are converted to 3 before migration (#49614)

* update how to docs to convert plugins to airflow3

* Update airflow-core/docs/installation/upgrading_to_airflow3.rst

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

* update docs

---------

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

amoghrajesh committed:
Update outdated doc-build selective checks (#49602)

jscheffl committed:
Make Edge3 Intergation Test DAG working in 2.10 (#49474)

amoghrajesh committed:
Add backcompat shim for get_current_context (#49630)

* Add backcompat shim for get_current_context

* use sdk!

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

---------

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

vincbeck committed:
Add read config permission to viewer role in `FabAuthManager` (#49581)

dstandish committed:
Remove merge conflict symbols from the airflow codebase (#49564)

Having these in the codebase results in false positives every time you
search for merge conflicts in the code.  And that's annoying.

jedcunningham committed:
Better upgrade docs about flask/fab plugins in AF3 (#49628)

The support we added isn't a "workaround", its absolutely intentional.
It's just deprecated :)

jedcunningham committed:
Update plugin documentation for Airflow 3 plugins (#49632)

This removes examples for FAB/Flask Airflow 2 style plugins, and adds an
section about the compatibility later in the FAB provider.

aritra24 committed:
feat (airflowctl): Shortcircuit command call on error (#49438)


* Moved error check up to ctl level

* More granular exceptions

dependabot[bot] committed:
Bump stylelint (#49638)

Bumps the fab-ui-package-updates group with 1 update in the /providers/fab/src/airflow/providers/fab/www directory: [stylelint](https://github.com/stylelint/stylelint).


Updates `stylelint` from 16.18.0 to 16.19.0
- [Release notes](https://github.com/stylelint/stylelint/releases)
- [Changelog](https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/stylelint/stylelint/compare/16.18.0...16.19.0)

---
updated-dependencies:
- dependency-name: stylelint
  dependency-version: 16.19.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: fab-ui-package-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bugraoz93 committed:
Fix SnowflakeSqlApiHook backwards compatibility for get_oauth_token method (#49482)

* Fix SnowflakeSqlApiHook backwards compatibility

* Update providers/snowflake/src/airflow/providers/snowflake/hooks/snowflake_sql_api.py

---------

Co-authored-by: Elad Kalif <45845474+eladkal@users.noreply.github.com>

guan404ming committed:
Add count to Stats Cards in Dashboard (#49519)

* AIP-38: enhence stats in dashboard

* fix: use semantic tokens

Co-authored-by: Brent Bovenzi <4600967+bbovenzi@users.noreply.github.com>

* fix: apply suggestions from code review

Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>

* fix: replace flex with hstack

* fix: add missing icon

Co-authored-by: Brent Bovenzi <4600967+bbovenzi@users.noreply.github.com>

---------

Co-authored-by: Brent Bovenzi <4600967+bbovenzi@users.noreply.github.com>
Co-authored-by: Brent Bovenzi <brent.bovenzi@gmail.com>

kaxil committed:
Update default Airflow version in Dockerfile to 3.0.0 (#49626)

3.0.0 is out :tada:

potiuk committed:
Prepare documentation to release common.compat 1.6.1 (#49624)

potiuk committed:
Fix missing version substitution in docs (#49640)

The separation of docs missed the substitution to replace version
in some documentation.

Fixes: #49622

jsjasonseba committed:
feature(ui): Make entire task box clickable to select the task in Air… (#49299)

* feature(ui): Make entire task node clickable to select the task

* Add z-index to TogglePause for clickability

* Remove link component

* Remove unnecessary onClick handler from LinkOverlay

* Forward ref from LinkOverlay

tirkarthi committed:
Remove extra slash so that the runs tab is selected. (#49600)

potiuk committed:
Fix min-sqlalchemy tests (#49636)

The #49452 broke min-sqlalchemy tests by adding asyncio as
dependency.

kaxil committed:
Chart: Make Airflow 3.0.0 default Airflow version (#49593)

guan404ming committed:
AIP-38: enhance layout for `RunBackfillForm` (#49609)

* feat: enhence layout for backfillForm

* fix: use chakra's `Field` component

Co-authored-by: Brent Bovenzi <4600967+bbovenzi@users.noreply.github.com>

---------

Co-authored-by: Brent Bovenzi <4600967+bbovenzi@users.noreply.github.com>

gopidesupavan committed:
Update standard provider doc operators in core operators-and-hooks-ref.rst (#49401)

* move standard provider doc references from core area

* Remove standard provider reference doc link in core

* Update operators-and-hooks-ref.rst with standard provider docs

* Add rst file ref to how-to-guide section

tirkarthi committed:
Update docs since --tree flag is tasks list command removed in Airflow 3. (#49623)

tirkarthi committed:
Add autorefresh to health section for live updates. (#49645)

gopidesupavan committed:
Fix test_roundtrip_provider_example_dags (#49647)

bbovenzi committed:
Fix Dag Code text selection bg (#49649)

potiuk committed:
Upgrade breeze deps (#49656)

potiuk committed:
Fix wrong link in standard provider yaml file (#49660)

potiuk committed:
Remove .dev0 suffix from airflow version (#49621)

dstandish committed:
Remove some lingering subdag references (#49663)

guan404ming committed:
Add external logs link (#49475)

* feat: support external log

* test: add test and dataModel

Co-Authored-By: LIU ZHE YOU <68415893+jason810496@users.noreply.github.com>

* refactor: test structure

Co-Authored-By: LIU ZHE YOU <68415893+jason810496@users.noreply.github.com>

* fix: airflowctl type in static check

* fix: api and ui

Co-authored-by: pierrejeambrun <pierrejbrun@gmail.com>

* fix: miss rel and coding style

* fix: replace a with `Link`

Co-authored-by: Brent Bovenzi <4600967+bbovenzi@users.noreply.github.com>

---------

Co-authored-by: LIU ZHE YOU <68415893+jason810496@users.noreply.github.com>
Co-authored-by: pierrejeambrun <pierrejbrun@gmail.com>
Co-authored-by: Brent Bovenzi <4600967+bbovenzi@users.noreply.github.com>

rawwar committed:
fix a few sqlalchemy deprecation warnings (#49477)

* fix sqlA warning in taskinstance.py

* fix deprecation in migration: 0059

* fix few more deprecations

* remove usage of _mapping

* run pre-commit to update erd

kaxil committed:
Docs: Use `get_current_context` from Task SDK (#49654)

The one in Standard Operator is just for Back compat

potiuk committed:
Switch to built-in dry-run/verbose commands for breeze release commands (#49616)

The release commands were using special "if CI" case which has been
pretty problematic to test and run - instead we switch to the built-in
dry-run/verbose functionality which will produce cleaner code.

amoghrajesh committed:
Extract logic to redirect on login/logout to a method (#49604)

potiuk committed:
Remove the "dev0" usage from the CI (#49665)

When we are building providers and airflow-core/task-sdk in our CI,
we stop using `.dev0` suffix - instead, when needed we use the
`--skip-tag-check` when building the providers - which should
eventually automatically handle building chicken-egg providers without
having to spcify and configuring them.

We want to first to make it work now without removing the "chicken-egg"
provider configuration. But the next step will be using the feature
where the chicken-egg providers configuration will be removed.

potiuk committed:
Remove the retries script - we do not use it any nore (#49687)

gopidesupavan committed:
Add S3 docs publishing scripts (#47814)

* Add s3 docs publishing scripts

* Add stg s3 bucket to test

* resolve conflicts

* Resolve vincent comments

* Rebase

* Update AIRFLOW_ROOT_PATH fix

* Resolve conflicts

ashb committed:
Make LatestOnlyOperator work for default data-interval-less DAGs (#49554)

Remove the check/don't skip logic whe the data interval is zero-wdith. Even if
a DAG doesn't have the concept of a data-interval (i.e. it is zero width) it
still is logically consistent for it to have to concept of latest or not, so
we now only compare against the end date of the interval.

(And a few drive-by refactors too, `context["task"]` is `self`,
`context["dag"]` is `self.dag`)

eladkal committed:
bugfix: Add handling for unhealthy state in `AzureContainerInstancesOperator` (#49688)

* bugfix: Add handling for unhealthy state in `AzureContainerInstancesOperator`

kacpermuda committed:
feat: Add DAG versioning information to OpenLineage events (#48741)

insomnes committed:
Simplify logic flow of initialize_airflow_tests fixture (#49461)

* refactor(tests): simplify initialize_airflow_tests

- separate db and kerberos initialization to functions
- remove double import of `initial_db_init` and import only if it
  is needed
- utilize guard clauses
- use pathlib.Path in lock file operations
- unify message wording

* fix: switch -> flag

ahmadtfarhan committed:
Create Apache TinkerPop provider (#47446)

* tests passed

* narrow down CI

* fix host

* add drill for testing

* add docker inspect

* create a funtion

* minor changes

* change the inspect

* add status condition

* change host to zeros

* rewrite the pipeline

* push image and reuse

* remove tag

* rename tag

* change to locally pushed image

* add sha to image name

* add sha to image name

* add debug and login

* change token name

* change token name

* remove login

* remove login

* remove login

* move token to hight level

* copy from ci

* change to secrets

* test workflow

* test workflow

* rerun on CI login

* rerun on CI login

* logout

* revert logout

* revert login

* remove cleanup

* readd cleanup

* change docker pass

* add login to action

* move up

* chage token

* one job

* one job

* remove action

* delete if condition

* rename image

* fix tar name

* fix tar name

* fix tar name

* remove status

* remove stdin

* add logs and hostname

* removed restart

* add chmod

* create entry sh

* create entry sh

* add status

* remove failure

* add user

* revert back all ci yamls

* revert back shell script

* fix tinkerpop

* change back to gremlin in the providers list

* change docs

* change docs

* add conn_name_attr

* move system test

* change to cap

* change to 1.0.0

* fix sh

* removed serializer and some changes

* fixing docs

* update provider

* change to 2.9.0 and add spell check

* ran breeze release management

* add doc strings

* add asterisk

* moved operators doc

* fixed docs

* fixed pyproject

* minor change to docs

* add conf.py

* fixed docs

* fix toml

* changed to 2.10

* remove fab from tinkerpop

* fix prov info

* add close method

* fix integration

* change gremlin host

* change gremlin host back

* remove async

* remove package

* fix pyproject

* fix test

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

potiuk committed:
Update dev docs about merging PRs for Airflow 3.0 (#49701)

eladkal committed:
Prepare docs for Apr ad hoc release of fab and common.compat providers (#49690)

ashb committed:
Tweak Execution API OpenAPI spec to improve code Generation (#49700)

This change improves the generated code using the OpenAPI schema:

- `anyOf` to `oneOf` so that generated clients don't try _all_ schemas, but
  stop after the first
- Fix the header pattern to make it valid against the spec
- Use the function name as the default operation ID in the spec.

On this last part, this is a different implementation (but same effect) to how
we do operation name in the Core API. I will change CoreAPI to use this same
method in a separate PR, but I didn't want to touch both APIs in this PR for
clarity of git history

potiuk committed:
Improve caching strategy for inventory cache (#49693)

* Improve caching strategy for inventory cache

Inventory cache is used to decrease the potential failures coming
from not-available (Tempoararily) external inventories of intersphinx.

This includes both - airflow package inventory and external intersphinx
docs as well.

We are more concerned about availability of third-party inventories
than ours, so we should use the cache mostly for the external packages
rather than for ours and we should refresh it every time any dependency
changes.

The caching strategy implemented originally is to keep inventories for
12 hours, but this might cause 12 hours of failures in CI since we are
using latest cached inventories for stability.

This change adds --refresh-airflow-inventories flag and uses it in
the CI to always refresh airflow inventories and leave the other
cached inventories.

* fixup! Improve caching strategy for inventory cache

abhishekbhakat committed:
Rename operation IDs for task instance endpoints to include map index… (#49608)

* Rename operation IDs for task instance endpoints to include map index for openapi spec validator

* refactor: Update TaskInstanceService methods and types to use map index for dependencies

* Add OpenAPI validation to FastAPI spec generation script

* refactor: streamline OpenAPI validation process by removing redundant exit checks

potiuk committed:
Confgure "suggest a change" links properly (#49702)

After moving the docs, the links to "Suggest a change" for airflow,
docker-compose and helm-chart lead to a wrong URL. This PR fixes it.

We also do not use analytics any more so configuration for it is not
needed any more.

Fixes: #49694

tirkarthi committed:
Check for task concurrency parameters inside encoded object & partial_kwargs to handle dynamic tasks. (#49708)

Closes #49696

rawwar committed:
use NonNegativeInt for backfill_id (#49691)

* use NonNegativeInt for backfill_id

* fix aws failing test

* update BackfillResponse

pierrejeambrun committed:
Display plugin import errors in the UI (#49643)

* Display plugin import errors in the UI

* Update following code review

* Handle user without plugin access

bugraoz93 committed:
feat (airflowctl): Allow adding the same commands with custom action commands (#49674)

jason810496 committed:
Fix k8s test: scheduler crash when using LocalExecutor (#49677)

amoghrajesh committed:
Better handle safe url redirects in login form for SimpleAuthManager (#49697)

pierrejeambrun committed:
Improve AirflowRouter (#49712)

ashb committed:
Fix infinite redirect caused by mistakenly setting token cookie as secure (#49721)

`api/ssl_cert` has a default value of "" so `has_option` is always returning
true, so we have to find in a slightly more complex way of telling if this
setting is turned on.

I have also updated the Simple security manager to look at X-Forwarded-Proto
so that if there is a reverse proxy in front (and if Airflow is configured to
trust that) then it sets the secure flag

guan404ming committed:
fix: PlugiuImportErrors assertion (#49719)

ashb committed:
Fix infinite redirect in FAB AuthManager caused by mistakenly setting token cookie as secure (#49724)

`api/ssl_cert` has a default value of "" so `has_option` is always returning
true, so we have to find in a slightly more complex way of telling if this
setting is turned on.

Provider version of #49721

rawwar committed:
include jti when generating jwt token (#49531)

* include jti when generating jwt token

* update tests to check if jti exists in token

ramonvermeulen committed:
feat: add `polling_period_seconds` and `timeout_seconds` to the template fields of `CloudRunExecuteJobOperator` (#49704)

dheerajturaga committed:
Fix Typo caused by #49721 (#49736)

kaxil committed:
Bump Task SDK to `1.1.0` (#49590)

Since 1.0.0 of it is out, let's make main target 1.1.0

MaksYermak committed:
Remove on_finish_action from template fields (#49637)

MaksYermak committed:
Create operators for working with Ray for VertexAI (#47340)

rawwar committed:
add SimpleAllAdminMiddleware to allow api usage without auth header in request (#49599)

* add SimpleAllAdminMiddleware

* add tests

* update test

* update test name

* refactor test

* sort test order

* hardcode endpoints

* use session

* db_test mark

* iterate over app.routes

* use fewer endpoints

* update comment

* update scope

dstandish committed:
Remove reference to root_dag_id in dagbag and restore logic (#49668)

Also I think that we used to remove the dag in queston from the dagbag when expired.

dependabot[bot] committed:
Bump openlineage-airflow from 1.31.0 to 1.32.0 (#49739)

Bumps openlineage-airflow from 1.31.0 to 1.32.0.

---
updated-dependencies:
- dependency-name: openlineage-airflow
  dependency-version: 1.32.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dstandish committed:
Clear out the dag code and serialized_dag tables on 3.0 upgrade (#49563)

This will discard the v1 serdags and let them be reserialized after new dag processor starts up.

Rather than go through the trouble of migrating the data for serialized dag and dag code, we can simply delete it and let it be regenerated after upgrade / downgrade.

Why does this make sense?

Prior to airflow version 3, both serialized_dag and dag_code would have been deleted every time the dag was reprocessed.  So, it was always ephemeral in 2.x.  And we typically did a `airflow dags reserialize` on upgrade.

So this is just deleting it one more time and reserializing it one more time on the way to 3.0, after which we we _don't_ delete everything with each run of dag processor.

There's little value in migrating the data when it can just be regenerated.

Similarly, when going back down to airflow 2.x from 3.0, rather than migrating the data, just delete it.  Because it will be regenerated in 2.x, and the PKs don't allow more than one version anyway.

dependabot[bot] committed:
Bump react-router and react-router-dom (#49742)

Bumps [react-router](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router) to 7.5.2 and updates ancestor dependency [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom). These dependencies need to be updated together.


Updates `react-router` from 7.4.0 to 7.5.2
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router@7.5.2/packages/react-router)

Updates `react-router-dom` from 7.4.0 to 7.5.2
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router-dom@7.5.2/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router
  dependency-version: 7.5.2
  dependency-type: indirect
- dependency-name: react-router-dom
  dependency-version: 7.5.2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Address PR comments, add newsfragment

kaxil committed:
Make default execution server URL be relative to API Base URL (#49747)

The execution API server URL is now more flexible and configurable:

- Remove hardcoded default URL "http://localhost:8080/execution/"
- Add fallback mechanism to construct URL from base_url config
- Default is now `"{BASE_URL}/execution/"` where BASE_URL comes from `api.base_url` config

vincbeck committed:
Add authentication section in Rest API documentation (#49745)

Commit:
trigger build, empty commit

kaxil committed:
Remove unused configurations (#49749)

gopidesupavan committed:
Set execution api url in breeze config environment (#49753)

* Set execution api url in breeze config environment

* Set execution api url from breeze shell params

dstandish committed:
Remove old dynamic attr accessing re dag resource in FAB provider (#49669)

This looks like it used to be a backcompat thing.  But now the permissons module is included in fab provider, so the `if getattr` logic always evaluates true.  So the else block does nothing.  So we can remove it and remove this `_resource_name` helper.  Additionally, I remove some `root_dag_id` logic that is also obsolete, since min provider version for fab is 3.0.0.

guan404ming committed:
fix: `get_uri` for oracle (#49533)

* fix: get_uri for oracle

Co-Authored-By: Christopher Jones <christopher.jones@oracle.com>

* test: remove invalid test

Co-Authored-By: Christopher Jones <christopher.jones@oracle.com>

---------

Co-authored-by: Christopher Jones <christopher.jones@oracle.com>

dstandish committed:
Add bundle_version to DagRun response (#49726)

The field was missing.

Note on relationship to #49007...

In #49007 the concern was raised about the DagVersion record not pointing to the current bundle version when the bundle version changed but dag structure did not.

The user assumed that this meant the dag run was associated with the wrong bundle version. Adding the dag run bundle version will let the user see that that the associaton is correct.

Users may now see two "bundle versions" in the dag run response.  But the reason for this can be understood.  If the DagVersion record points to an older bundle version, it is because the dag structure did not change when the code changed.  But since the scheduler just cares about dag structure, it doesn't matter, and it still tells the truth -- the version of the dag version is the bundle version which caused that revision.

closes https://github.com/apache/airflow/issues/49007

Commit:
Fix imports

dheerajturaga committed:
Minor doc fix in edge_executor (#49755)

Commit:
Revert "Fix imports"

This reverts commit 6fecc928636625396141a543707a9aaeac446547.

amoghrajesh commented on airflow-core/src/airflow/api_fastapi/core_api/openapi/v1-rest-api-generated.yaml:
```diff
@@ -8337,11 +8337,6 @@ components:
-          anyOf:
-          - type: string
-          - type: 'null'
-          title: Bundle Version
```
Unrelated right?

amoghrajesh commented on providers/fab/src/airflow/providers/fab/auth_manager/fab_auth_manager.py:
```diff
@@ -638,6 +651,23 @@ def _get_user_permissions(user: User):
+            return self.appbuilder.get_session.scalar(
+                select(DagModel.dag_id, DagModel.root_dag_id).where(DagModel.dag_id == dag_id).limit(1)
+            )
+        return dag_id
```
Same with this file

prabhusneha replied:
Yes, unrelated. Give me a few mins, fixing it

Commit:
Fix
